modules/src/crypto/mod.rs

```diff
@@ -8,10 +10,40 @@ pub(crate) fn keccak256(input: &str) -> [u8; 32] {
     result.try_into().unwrap()
 }
 
+/// This function is used to recover the public key from the signature.
+/// The signature needs to be in libsecp256k1 format.
+/// hash is the hash of the message that was signed.
+/// recovery_id is the recovery id of the signature (0 to 3).
+#[allow(dead_code)]
+fn ecrecover(hash: &[u8; 32], signature: &[u8; 64], recovery_id: u8) -> Result<PublicKey, Error> {
+    let message = libsecp256k1::Message::parse_slice(hash)?;
+    let recovery_id = libsecp256k1::RecoveryId::parse(recovery_id)?;
+    let signature = libsecp256k1::Signature::parse_standard_slice(signature)?;
+    let public_key = libsecp256k1::recover(&message, &signature, &recovery_id)?;
+    Ok(public_key)
+}
+
+fn verify_signature(hash: Bytes, signature: Bytes, public_key: odra::types::PublicKey) -> bool {
```

Comment:
This should be part of `contract_env`. This way you'll have implementations for both mock and casper where they should be and this module could be removed.

Reply:
This module stays due to keccak that is used in AccessControl module.